libs/agno/agno/vectordb/chroma/chromadb.py

```diff
@@ -18,6 +21,12 @@
 from agno.vectordb.base import VectorDb
 from agno.vectordb.distance import Distance
 
+# Type variable for generic function return types
+T = TypeVar("T")
+
+# Thread pool executor for async operations
+thread_pool_executor = concurrent.futures.ThreadPoolExecutor(max_workers=10)
```

Comment:
This adds unnecessary overhead. Only needed at scale. We can get away with plain threading.

Reply:
got it, let me change